### PR TITLE
Adds some neat tiles and walls for use on lavaland

### PR DESCRIPTION
**Pull request by ChangelingRain**
@Ergovisavi you were talking about bosses making arenas I think you might possibly want these for that.
Walls, normal and red/hot, not pictured since I'm a donk.
Sprites, besides ones I edited, are credit Flurrymcflurry who I don't know the github account of.

These are mainly intended for mapper use, and are set up in the sprite file accordingly.
Most, if not all, are directional, and there are two styles not pictured in those images, plus three more styles with a more red color palette.

These prevent you from placing survival pods on them, since it'd look weird, and keep you from falling into chasms/burning on lava.
Which might look weird with some dirs/low amounts of tiles on a lava/chasm tile, but that's noooot really my problem and making a check for a "proper amount" of tiles on a lava/chasm tile seems like bad code.

### CODE REVIEW

**Comment by DoctoreMad** [comment]
>all this non-pictured shit

aaaaaaaaaaaaaaaaaaaa
I like what I see so far but its like 20% of the PR.

**Comment by ChangelingRain** [comment]
There is so much stuff and doing it ingame is a pain in the butt so you're probably not getting more shots

**Comment by Supermichael777** [comment]
smashed tiles floating in the air, the magi-tech version of a rickety bridge.

**Comment by Chronitonity** [comment]
>not pictured

amazing tbh fam

File: code/modules/ruins/objects_and_mobs/necropolis_gate.dm [hunk 1]
```diff
@@ -60,3 +60,168 @@
...
+	tile_random_sprite_max = 2
+
+/obj/structure/stone_tile/surrounding_tile
+	name = "stone surrounding tile"
+	icon_state = "pristine_surrounding_tile1"
+	tile_key = "pristine_surrounding_tile"
+	tile_random_sprite_max = 2
+
+//cracked stone tiles
+/obj/structure/stone_tile/cracked
```
Comment (RemieRichards) [inline code/modules/ruins/objects_and_mobs/necropolis_gate.dm]: crossing a cracked tile should make it animate into the floor (pixel_z iirc) and then delete itself.
you know, classic video game stuff.

Comment (ChangelingRain) [inline code/modules/ruins/objects_and_mobs/necropolis_gate.dm]: Only on chasms and lava, though.

**Comment by ChangelingRain** [comment]
this isn't even all of them and it took like 14 minutes to set up

**Comment by ChangelingRain** [comment]
bonus shot so you can see just how cool(hot) these are

**Comment by GunHog** [comment]
Make the tiles move and you have yourself a platformer going!

**Comment by KorPhaeron** [comment]
Neat

**Comment by KorPhaeron** [comment]
Should replace the necropolis bridge with these

**Comment by FuryMcFlurry** [comment]
oh hey i made the tiles, i might plan on continuing adding more objects. ill scrap the red tiles.

**Comment by ChangelingRain** [comment]
Those are pretty clearly using the red tile palette, don't forget to recolor them for the standard brown.

**Comment by FuryMcFlurry** [comment]
image
already ahead of you.

**Pull request merged**